crates/sui-types/src/object.rs

```diff
@@ -689,6 +690,30 @@ impl Object {
         ObjectDigest::new(default_hash(self))
     }
 
+    pub fn is_coin(&self) -> bool {
+        if let Some(move_object) = self.data.try_as_move() {
+            move_object.type_().is_coin()
+        } else {
+            false
+        }
+    }
+
+    pub fn as_coin_maybe(&self) -> Option<Coin> {
```

Comment:
Instead of returning a `Coin` (which requires an expensive deserialization), I would suggest exposing `get_balance_maybe(&self) -> Option<u64>` or `get_balance_unsafe(&self) -> u64` that use `MoveObj::get_balance_unsafe`. Any caller of this is just going to look at the balance anyway, so this should be cheaper + more direct.

Reply:
this is a good idea, I will follow up separately.

Reply:
added a TODO